### PR TITLE
FreeRDP : switch to cmake-1.1 PG plus fixes and cleanup

### PR DESCRIPTION
**Pull request by RJVB**
This makes the port use the cmake 1.1 PortGroup and applies various
fixes and cleanups:
- use `use_xcode` (because of calling ibtool during the build)
- don't link opportunistically to various other libraries
- provide variants to control the feature set:
* +codecs for additional compression codecs including libjpeg which
  could even be made a default (turbo variant preferred)
* +cups to allow printer redirection (could also be default since
  CUPS is a system commodity)
* +gssapi to allow GSSAPI authentication (important for newer MSWin
  versions?!)
* +pulseaudio to support audio via the PulseAudio daemon
* +docs to disable building the manpages, something I've seen fail
  for unknown reasons.

See: #47197

###### Type(s)

- [x ] bugfix
- [ x] enhancement
- [ ] security fix

###### Tested on

OS X 10.9.5

### CODE REVIEW

**Comment by RJVB** [comment]
>I also note that ibtool emits the following warning during build (although it builds and runs without problems as far as I can see).

I noticed the same or similar warning but had no idea what one could do about it (if anything). I also note that there is no .nib or .xib file among the installed content; can these resource files be embedded in a shared library nowadays?

File: net/FreeRDP/Portfile [hunk 1]
```diff
@@ -44,18 +48,73 @@ post-patch {
...
+    configure.args-append \
+                    -DWITH_JPEG=ON \
+                    -DWITH_LAME=ON \
+                    -DWITH_FAAD2=ON \
+                    -DWITH_FAAC=ON \
+                    -DWITH_SOXR=ON
+}
+
+variant gssapi description {build GSSAPI support} {
+    depends_lib-append \
```
Comment (cjones051073) [inline net/FreeRDP/Portfile]: Again, does this really need to be an optional variant ?

**Comment by RJVB** [comment]
On Thursday January 02 2020 16:00:53 Chris Jones wrote:

>Otherwise the configure command has both ON and OFF settings in it which is at best confusing but also relying on the implicit ordering.

You're absolutely right, I think I know why I overlooked this. Fixed.



>+variant codecs description {support JPEG, MP3, FAAD2, FAAC and SOXR codecs} {
>+    # prefer the much faster libjpeg-turbo
>
>Is this something most users would want ? If so it definitely should be a default variant, and even better not a variant at all, and just on by defsult.

JPEG probably (I presume it's used for the screen image transfer), so I enabled it always. The others I must admit I have no idea of, other than that they're off by default in the build system. If we assume that most users will have lame installed we can enable MP3 support too.

TBH, I had a very old FreeRDP installed, had to address the main issues covered here and then discovered after upgrading that I no longer have any MSWin machine that will allow me to "rdp in". I only decided to do this PR because I realised the port wouldn't build as it currently is AND is covered by the cmake PG ticket on Trac. The variants are just icing on the cake.

**Comment by cjones051073** [comment]
@dbevans when someone requests changes in a PR, I think it is normal to wait until that person has reviewed their requested changes before merging things.

File: net/FreeRDP/Portfile [hunk 2]
```diff
@@ -44,18 +48,73 @@ post-patch {
...
+                    port:lame \
+                    port:faad2 \
+                    port:faac \
+                    port:soxr
+    configure.args-append \
+                    -DWITH_JPEG=ON \
+                    -DWITH_LAME=ON \
+                    -DWITH_FAAD2=ON \
+                    -DWITH_FAAC=ON \
+                    -DWITH_SOXR=ON
```
[the remark this answers is not in the document]
Comment (cjones051073) [reply net/FreeRDP/Portfile]: Are you sure what you changed here works ?
Even if it does, I would have prefered to have multiple replace calls, with each having only two strings, for before and after. I think that would have been much clearer than what you have no, personnally.

**Comment by RJVB** [comment]
>@dbevans when someone requests changes in a PR, I think it is normal to wait until that person has reviewed their requested changes before merging things.

Esp. since Chris was right about a wrong application of the -replace feature.

Also: I'm pretty certain my copy had a path-style dependency on port:libjpeg-turbo, which is exactly why I used that kind of depspec. I don't know if the turbo variant has drawbacks in terms of image quality or features but for a remote desktop client it has the definite advantage of being a lot faster.

I'll see what I can do after merging.

**Pull request merged**